snakemake_executor_plugin_slurm/__init__.py

```diff
@@ -408,6 +408,13 @@ def cancel_jobs(self, active_jobs: List[SubmittedJobInfo]):
                 )
             except subprocess.TimeoutExpired:
                 self.logger.warning("Unable to cancel jobs within a minute.")
+            except subprocess.CalledProcessError as e:
+                msg = e.stderr.decode().strip()
+                if msg:
+                    msg = f": {msg}"
+                raise WorkflowError(
+                    f"Unable to cancel jobs with scancel (exit code {e.returncode}){msg}"
+                )
```

Comment:
I am unsure about the exit codes of slurm scancel. Do you know more? THis here would fail with a WorkflowError for anything except 0. In the docs there is nothing about exit codes. I recently received an exit code of 8 from scancel. And gemini says the following without being able to give me a source for it:

>     0: The job was successfully canceled.
    1: The job could not be found.
    2: The job is already in the completed state.
    3: The job is already in the canceled state.
    4: The job is already in the suspended state.
    5: The job is already in the pending state.
    6: The job is already running.
    7: The job is already in the requeued state.
    8: The job was successfully canceled.
    9: The job was successfully suspended.
    10: The job was successfully requeued.
    11: The job was successfully resumed.
    12: The job was successfully held.
    13: The job was successfully released.
    14: The job was successfully signaled.
    15: The job was successfully terminated.
    16: The job was successfully restarted.
    17: The job was successfully updated.
    18: The job was successfully requeued to a different partition.
    19: The job was successfully requeued to a different queue.
    20: The job was successfully requeued to a different priority.
    21: The job was successfully requeued to a different account.
    22: The job was successfully requeued to a different user.
    23: The job was successfully requeued to a different group.
    24: The job was successfully requeued to a different association.
    25: The job was successfully requeued to a different node.
    26: The job was successfully requeued to a different nodelist.
    27: The job was successfully requeued to a different feature.
    28: The job was successfully requeued to a different constraint.
    29: The job was successfully requeued to a different time.
    30: The job was successfully requeued to a different deadline.
    31: The job was successfully requeued to a different dependency.
    32: The job was successfully requeued to a different requeue priority.
    33: The job was successfully requeued to a different requeue time.
    34: The job was successfully requeued to a different requeue deadline.
    35: The job was successfully requeued to a different requeue dependency.